### ISSUE TITLE
Postconstruct async reslove

### ISSUE THREAD

**Issue by tlyau62** [tlyau62/UnityAddon3#14]
The following code may break the resolve stack, causing exception Circular dependency.
```c#
postconstruct {
    new Task {
        container.resolve<T>();
    }
}
```

Workaround is to move the dependency outside the new thread scope if the dependency is singleton bean.

**Comment by tlyau62** [comment]
Should not use IAsyncLocalFactory for the resolve stack.

**Comment by tlyau62** [comment]
Fix at commit f61750a64e53919f21f599d3be416d2cf22c2585
